### PR TITLE
Fix some build warnings

### LINKED ISSUE #713: Compiler warnings elimination
Starting from the creation of this task, #116 is a subtask of this task.

MSVC:
- [ ] MSBuild
  - [ ] Enable "warnings are errors" mode *(is it possible?)*
- [ ] Compiler
  - [x] Enable conformance mode
  - [x] Enable "warnings are errors" mode
  - [ ] Raise the warning level to 4
    - [x] Level 1
    - [x] Level 2
    - [ ] Level 3
    - [ ] Level 4
- [ ] Linker
  - [ ] Enable "warnings are errors" mode

GCC:
- CMake and `make`
  - [ ] Enable "warnings are errors" mode *(is it possible?)*
- [ ] Compiler
  - [x] Disable permissive mode
  - [ ] *Maybe enable [pedantic mode](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)*
  - [ ] Enable "warnings are errors" mode
  - [ ] Raise the warning level to 4
    - [ ] Level 1
    - [ ] Level 2
    - [ ] Level 3
    - [ ] Level 4
- [ ] Linker
  - [ ] Enable "warnings are errors" mode

For history:
- [ ] List all warnings elimination commits in this issue

Tools:
PVS-Studio
CppDepend (includes cppcheck, Clang-Tidy, Vera++)
ReSharper C++ (includes Clang and Clang-Tidy check plus it's own analysis)
Visual Assist (has it's own code inspection)

https://habr.com/ru/articles/75123/
https://habr.com/ru/companies/pvs-studio/articles/303664/

### PR DESCRIPTION
**Pull request by Masterkatze**

### CODE REVIEW

File: src/xrGame/script_game_object2.cpp [hunk 1]
```diff
@@ -153,7 +153,7 @@ void CScriptGameObject::Hit(CScriptHit* tpLuaHit)
     SHit HS;
     HS.GenHeader(GE_HIT, object().ID()); //	object().u_EventGen(P,GE_HIT,object().ID());
     THROW2(tLuaHit.m_tpDraftsman,
-        "Where is hit initiator??!"); //	THROW2			(tLuaHit.m_tpDraftsman,"Where is hit initiator??!");
+        "Where is hit initiator?!"); //	THROW2			(tLuaHit.m_tpDraftsman,"Where is hit initiator?!");
```
Comment (Xottab-DUTY) [inline src/xrGame/script_game_object2.cpp]: Don't touch GSC comments.

File: src/xrGame/Actor.cpp [hunk 2]
```diff
@@ -83,7 +83,7 @@ const float respawn_auto = 7.f;
 extern float cammera_into_collision_shift;
 extern int g_first_person_death;
 
-string32 ACTOR_DEFS::g_quick_use_slots[4] = {NULL, NULL, NULL, NULL};
+string32 ACTOR_DEFS::g_quick_use_slots[4] = {{'\0'}, {'\0'}, {'\0'}, {'\0'}};
```
Comment (Xottab-DUTY) [inline src/xrGame/Actor.cpp]: If I'm not mistaken, this should be enough:
```suggestion
string32 ACTOR_DEFS::g_quick_use_slots[4] = {};
```

File: src/xrGame/script_game_object2.cpp (hunk 1 above)
Comment (Masterkatze) [reply src/xrGame/script_game_object2.cpp]: `??!` is being treated as trigraphs and generates warning because it's deprecated.

Comment (Xottab-DUTY) [reply src/xrGame/script_game_object2.cpp]: Oh...

Comment (Xottab-DUTY) [reply src/xrGame/script_game_object2.cpp]: Please, return original emotion and importance of the assert in the text and just disable ridiculous warning via key.

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**